bin/client/main.rs

```diff
@@ -108,6 +109,51 @@ async fn main() -> std::result::Result<(), ClientError> {
             .await?;
             println!("{}", toml::to_string_pretty(&label)?);
         }
+        SubCommand::CreateKey(create_opts) => {
+            let dir = create_opts.secret_file.unwrap_or_else(|| {
+                default_config_dir()
+                    .unwrap_or_else(|| PathBuf::from("."))
+                    .join("secret_keys.toml")
+            });
+            println!("Writing keys to {}", dir.display());
+
+            match tokio::fs::metadata(&dir).await {
+                Err(e) if matches!(e.kind(), std::io::ErrorKind::NotFound) => {
+                    println!("File {} does not exist. Creating it.", dir.display());
+                    let mut keyfile = SecretKeyFile::default();
+                    let newkey = SecretKeyEntry::new(
+                        create_opts.label,
+                        vec![bindle::SignatureRole::Creator],
+                    );
+                    keyfile.key.push(newkey);
+                    keyfile
+                        .save_file(dir)
+                        .await
+                        .map_err(|e| ClientError::Other(e.to_string()))?;
+                }
+                Ok(info) => {
+                    if !info.is_file() {
+                        eprint!("Path must point to a file.");
+                        return Err(ClientError::Other(
+                            "Keyfile cannot be directory or symlink".to_owned(),
+                        ));
+                    }
+                    let mut keyfile = SecretKeyFile::load_file(dir.clone())
+                        .await
+                        .map_err(|e| ClientError::Other(e.to_string()))?;
```

Comment:
I missed this before, so I am not going to block, but there is a `ClientError:Io` type. So you should be able to just do `e.into()`. I can't remember how smart the `?` operator is, but it might even be able to convert without the mapping

Reply:
Since load_file() can cause IO, Dalek, and TOML/serde errors, it looks like I have to leave this as-is. But I did find another place where I can convert to a ClientError::Io.

Reply:
We might want a new class of error eventually for Signature stuff. But obviously not right now